### ISSUE TITLE
non-input Date variable

### ISSUE THREAD

**Issue by whitemuu** [Waboodoo/HTTP-Shortcuts#152]
Would you consider provide a checkbox option like "silence with now" in Data input variable ?

When checked, instead of popping up the input dialog while execution, use now silently.

**Comment by stevenjohnk** [comment]
Can't you use constant variable for your purpose? I think can help you, but also I am suggesting to my lovely developer to think more on variables for more freedom. Thanks

**Comment by whitemuu** [comment]
I want the current time, and my request consumer(IFTTT actually) isn't programmable, so I hafta provide  it in HTTP-shortcuts. constant variable, for what I know, doesn't serve the purpose.

**Comment by stevenjohnk** [comment]
Ok, dev may work on it, auto online json import(not ready implemented) can serve your purpose , where you can prepare value on your server side and parse it in variable in real time

**Comment by Waboodoo** [comment]
You can try using a variable of type "constant", and then add a "Pre-Request Action" to your shortcut in which you set the value of that constant to the current date.

See Examples in http://waboodoo.ch/http-shortcuts/scripting

```js
const currentDate = (new Date()).toDateString();
setVariable('today', currentDate);
```

**Comment by whitemuu** [comment]
YES! me dug too little.

**Comment by whitemuu** [comment]
Hi @Waboodoo , I noticed that constants I referenced in request JSON body are values I set in the Pre-Request Actions of lastly fired request. Is this a desired behaviour? why?

```
const paddingZero = i => i < 10 ? '0' + i : i;
const t = new Date();
const datetime = '' +
      paddingZero(t.getMonth() + 1) + '/' +
      paddingZero(t.getDate()) + '/' +
      t.getFullYear() + ' ' +
      paddingZero(t.getHours()) + ':' +
      paddingZero(t.getMinutes()) + ':' +
      paddingZero(t.getSeconds());
const yearmonth = '' +
      t.getFullYear() + '-' +
      paddingZero(t.getMonth() + 1);

setVariable("yearmonth", yearmonth);
setVariable("datetime", datetime);
```

**Comment by Waboodoo** [comment]
This is a bug. Thanks for reporting. I will look into it.

**Comment by Waboodoo** [comment]
This bug was fixed in version 1.25.0
